python/opae.admin/opae/admin/sysfs.py

```diff
@@ -599,8 +601,39 @@ def aer(self, values):
         except CalledProcessError as err:
             self.log.warn('error setting aer: %s', err)
 
+    def data_link_layer_active(self):
+        """data_link_layer_active Return true if PCIe data link is active.
+
+        Returns:
+            Bit 13 of the PCIe link status register.
+
+        Notes:
+            This relies on calling 'setpci' and will log an exception if an
+            error was encountered while calling 'setpci'.
+        """
+        try:
+            ls = int(call_process(self._link_status_cmd), 16)
+            return (ls >> 13) & 1
+        except CalledProcessError as err:
+            self.log.warn('error reading link status: %s', err)
+
+    def wait_for_data_link(self, target_state):
+        """wait_for_data_link Wait until data link active state matches
+                              target_state.
+        """
+        # Bridges are not required to implement the link layer active bit.
+        # Time out after a generous 5 seconds, assuming the target state
+        # is reached.
+        for i in range(5):
+            if self.data_link_layer_active() == target_state:
+                break
+            time.sleep(1)
+
+        # The spec requires at least 100ms delay after the link change.
+        time.sleep(0.25)
```

Comment:
> Would 0.1s pass your testing, too, or is the larger value still needed for certain implementations?

Most scripts add a little extra margin for safety since there is no ACK, especially in cases where latency doesn't matter.